apps/picsa-tools/monitoring-tool/src/app/components/form-item/form-item.component.ts

```diff
@@ -33,25 +37,80 @@ export class FormItemComponent implements OnInit, OnDestroy {
     failed: { ...STATUS_ICONS.failed, value: 0 },
   };
 
-  private componentDestroyed$ = new Subject();
+  public isProcessing = signal(false);
 
-  constructor(private service: MonitoringToolService, private cdr: ChangeDetectorRef) {}
+  private componentDestroyed$ = new Subject<void>();
+
+  constructor(
+    private service: MonitoringToolService,
+    private cdr: ChangeDetectorRef,
+    private dialog: MatDialog,
+    private router: Router,
+    private snackBar: MatSnackBar
+  ) {}
 
   public get syncStatus() {
     return hashmapToArray(this.syncStatusMap, 'id');
   }
 
-  ngOnInit() {
+  // Check if form is locked
+  public get isLocked(): boolean {
+    return !!this.form.access_code && !this.form.access_unlocked;
+  }
+
+  ngOnInit(): void {
     this.subscribeToSubmissionSummary();
   }
 
-  ngOnDestroy() {
-    this.componentDestroyed$.next(true);
+  ngOnDestroy(): void {
+    this.componentDestroyed$.next();
     this.componentDestroyed$.complete();
   }
 
+  // Handle form click when locked
+  public async handleFormClick(event: Event): Promise<void> {
+    if (this.isLocked) {
+      event.preventDefault();
+      event.stopPropagation();
+
+      this.isProcessing.set(true);
+
+      const dialogRef = this.dialog.open(AccessCodeDialogComponent, {
+        width: '350px',
```

Comment:
+1 - I think this should work well, as far as I'm aware the smallest device size I typically come across is around 375px so should still fit.

The alternate approach is to set it as `100%` and use the content dialog to be a bit more responsive, e.g. including a container with a `max-width:...` and `margin: auto`, but for a simple dialog I think this works fine